src/Behat/Behat/Output/Node/EventListener/JUnit/JUnitOutlineStoreListener.php

```diff
@@ -56,25 +56,24 @@ public function listenEvent(Formatter $formatter, Event $event, $eventName)
     {
         $this->captureOutlineOnBeforeOutlineEvent($event);
         if ($event instanceof SuiteTested) {
-            setFormatterFileName($event, $formatter);
+            $formatter->fileName = $this->getFormatterFileName($event);
         }
 
         $this->printHeaderOnBeforeSuiteTestedEvent($formatter, $event);
         $this->printFooterOnAfterSuiteTestedEvent($formatter, $event);
     }
 
     /**
-     * Get FeatureNode from within Event
+     * Get FeatureNode from within Event and return the feature path name.
      *
      * @param Event $event
      */
-    private function setFormatterFileName(Event $event, Formatter $formatter) {
+    private function getFormatterFileName(Event $event) 
+    {
         foreach ($event->getSpecificationIterator() as $node) {
             if ($node instanceof FeatureNode) {
                 $path = pathinfo($node->getFile(), PATHINFO_FILENAME);
-                if (!empty($path)) {
-                    $formatter->fileName = $path;
-                }
+                return !empty($path) ? $path : NULL;
```

Comment:
If would `return` only if the path is not empty so it would try for the next `FeatureNode` in the iterator. And if not not empty path is found then it will return `null` by default.